interfaces/apparmor/spec.go

```diff
@@ -173,6 +174,92 @@ func isProbablyPresent(path string) bool {
 	return path == "/" || path == "/snap" || path == "/var" || path == "/var/snap" || path == "/tmp" || path == "/usr" || path == "/etc"
 }
 
+// chopTree takes a path and depth and returns two lists of apparmor path expressions.
+//
+// The returned lists of expressions are referred to as left and right.
+//
+// The left list describes directories at depth up to and including
+// assumedPrefixDepth and can be used to grant read permission to them (by
+// appending the string "r, " to each element). This corresponds to an
+// assumption about those directories being present in the system and being
+// just traversed. Note that depth is defined as the number of directories
+// traversed, including the root directory.
+//
+// The right list describes the remaining directories and the leaf entry
+// (either file or directory) and is somewhat subtle. At each depth level the
+// expression describes all the files and directories at that level. Coupled
+// with the string "rw ," it can be used to create a rule that allows write
+// access to any file therein, but not deeper.
+//
+// For example, with path: "/foo/bar/froz/baz" and depth 3 the result is:
+// []string{"/", "/foo/", "/foo/bar/"} and []string{"/foo/*", "/foo/bar/*",
+// "/foo/bar/froz/*"}. Coupled with the aforementioned constants this
```

Comment:
This comment doesn't coincide with the below rules. With the generated rules below I expected the return value to be: `[]string{"/", "/foo/", "/foo/bar/"}` and `[]string{"/foo/bar/*", "/foo/bar/*/", "/foo/bar/froz/*", "/foo/bar/froz/*/"}` (ie, the right hand side (RHS) in your comment has `"/foo/*"`, but that isn't represented in the rules below). I see in the testsuite updates below that you use exactly what I expected, so this is only a comment update.

Reply:
This is done